pkg/runctl/log/custom/pipelinerun_accessor.go

```diff
@@ -0,0 +1,48 @@
+package custom
+
+import (
+	"github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1"
+	yaml "gopkg.in/yaml.v3"
+)
+
+// LoggingDetailsProvider extracts details from a pipeline run to be added to the log
+type LoggingDetailsProvider func(run *v1alpha1.PipelineRun) []any
```

Comment:
I tink we don't need a new type here by just a type alias:

```suggestion
type LoggingDetailsProvider = func(run *v1alpha1.PipelineRun) []any
```